apps/web/vibes/soul/primitives/inline-email-form/index.tsx

```diff
@@ -21,11 +22,13 @@ export function InlineEmailForm({
   action,
   submitLabel = 'Submit',
   placeholder = 'Enter your email',
+  dismissLabel,
 }: {
   className?: string;
   placeholder?: string;
   submitLabel?: string;
-  action: Action<SubmissionResult | null, FormData>;
+  dismissLabel?: string;
+  action: Action<(SubmissionResult & { successMessage?: string }) | null, FormData>;
```

Comment:
Update state type to include `successMessage`, but I wonder if we should create a new type that can be used in other components.

Reply:
See my other comment about wrapping the submission result instead